test/sql/cast/string_to_integer_exponent_cast.test

```diff
@@ -0,0 +1,143 @@
+# name: test/sql/cast/string_to_integer_exponent_cast.test
+# description: String to Integer casts with Decimals and Exponents
+# group: [cast]
```

Comment:
I thought of another test where we create the minimal/max value and prefix it with `0.000...`, then construct the exponent to restore the value from the fraction

But that seemed to trigger an assertion
```sql
D select '0.170141183460469231731687303715884105727e39'::HUGEINT;
┌─────────────────────────────────────────────────────────────────┐
│ CAST('0.170141183460469231731687303715884105727e39' AS HUGEINT) │
│                             int128                              │
├─────────────────────────────────────────────────────────────────┤
│                         170141183460469231731687303715884105727 │
└─────────────────────────────────────────────────────────────────┘
D select '0.0170141183460469231731687303715884105727e40'::HUGEINT;
Error: INTERNAL Error: Assertion triggered in file "/Users/thijs/DuckDBLabs/duckdb/src/common/types/hugeint.cpp" on line 314: !(rhs.upper == 0 && rhs.lower == 0)
```